prov/util/src/util_atomic.c

```diff
@@ -163,15 +296,44 @@ size_t ofi_datatype_size(enum fi_datatype datatype)
 		ofi_complex_##type *r = (res);				\
 		OFI_UNUSED(src);					\
 		for (i = 0; i < cnt; i++)				\
-			r[i] = d[i];					\
+			op(type, d[i], r[i]);				\
```

Comment:
Should this be r[i] = op(...)?

Reply:
No, I have to use __atomic_load() instead of __atomic_load_n() so that it can support non-integer types. __atomic_load() sets the return parameter supplied as an argument.